File: dynamic/Maruko.Dynamic.Config/Internal/Domain/PageConfig.cs
﻿using System;
using System.Collections.Generic;
using System.Text;
using FreeSql.DataAnnotations;
using Maruko.Core.FreeSql.Internal;

namespace Maruko.Dynamic.Config
{
    [Table(Name = "dc_page_config")]
    public class PageConfig : FreeSqlEntity
    {
        [Column(Name = "pageId")]
        public long PageId { get; set; }
        [Column(Name = "url", StringLength = 200)]
        public string DataUrl { get; set; }
        [Column(Name = "isRow")]
        public bool IsRow { get; set; }
        [Column(Name = "isMultiple")]
        public bool IsMultiple { get; set; }
        [Column(Name = "rowWith")]
        public int RowWith { get; set; }
        [Column(Name = "fields",DbType = "text")]
        public string Fields { get; set; }
        [Column(Name = "buttons",DbType = "text")]
        public string Buttons { get; set; }
        [Column(Name = "functions",DbType = "text")]
        public string Functions { get; set; }
        [Column(Name = "isTableOperateRow")]
        public bool IsTableOperateRow { get; set; }
        [Column(Name = "IsTableCheckRow")]
        public bool IsTableCheckRow { get; set; }
    }
}

File: act/act-modules/act-modules-zero/src/main/java/com/act/modules/zero/internal/domain/PageConfig.java
package com.act.modules.zero.internal.domain;

import com.act.core.domain.BaseEntity;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

@Data
@TableName("dc_page_config")
@SuppressWarnings("all")
public class PageConfig extends BaseEntity<Long> {
    @TableField("pageId")
    private long pageId;
    @TableField("url")
    private String dataUrl;
    @TableField("isRow")
    private Boolean isRow = false;
    @TableField("isMultiple")
    private Boolean isMultiple = false;
    @TableField("rowWith")
    private int rowWith = 0;
    @TableField("fields")
    private String fields;
    @TableField("buttons")
    private String buttons;
    @TableField("functions")
    private String functions;
    @TableField("isTableOperateRow")
    private Boolean isTableOperateRow = false;
    @TableField("isTableCheckRow")
    private Boolean isTableCheckRow = false;
}
